beacon-chain/rpc/lookup/blocker.go

```diff
@@ -266,6 +270,30 @@ func (p *BeaconDbBlocker) Blobs(ctx context.Context, id string, indices []int) (
 		}
 	}
 
+	// Convert versioned hashes to indices if provided
+	indices := cfg.Indices
+	if len(cfg.VersionedHashes) > 0 {
+		// Build a set of requested versioned hashes for fast lookup
+		requestedHashes := make(map[string]bool)
+		for _, versionedHash := range cfg.VersionedHashes {
+			requestedHashes[string(versionedHash)] = true
+		}
+
+		// Create indices array in the order of KZG commitments in the block
+		indices = make([]int, 0, len(cfg.VersionedHashes))
+		for i, commitment := range commitments {
+			versionedHash := primitives.ConvertKzgCommitmentToVersionedHash(commitment)
+			if requestedHashes[string(versionedHash[:])] {
+				indices = append(indices, i)
+			}
+		}
+
+		// Verify all requested hashes were found
+		if len(indices) != len(cfg.VersionedHashes) {
+			return nil, &core.RpcError{Err: errors.New("versioned hash does not exist in given block"), Reason: core.NotFound}
```

Comment:
This would require writing some more code, but it would be nice to let the caller know which exactly which hash(es) were not found